packages/ember-glimmer/lib/ember-views/child-views-support.js

```diff
@@ -0,0 +1,73 @@
+/**
+@module ember
+@submodule ember-views
+*/
+import { Mixin } from 'ember-metal/mixin';
+import { get } from 'ember-metal/property_get';
+import { getOwner, setOwner, OWNER } from 'container/owner';
+
+export default Mixin.create({
+  init() {
+    this._super(...arguments);
+
+    /**
+      Array of child views. You should never edit this array directly.
+      Instead, use `appendChild` and `removeFromParent`.
+
+      @property childViews
+      @type Array
+      @default []
+      @private
+    */
+    this.childViews = [];
+    this.ownerView = this.ownerView || this;
+  },
+
+  appendChild(view) {
+    this.linkChild(view);
+    this.childViews.push(view);
+  },
+
+  destroyChild(view) {
+    view.destroy();
+  },
+
+  /**
+    Removes the child view from the parent view.
+
+    @method removeChild
+    @param {Ember.View} view
+    @return {Ember.View} receiver
+    @private
+  */
+  removeChild(view) {
+    // If we're destroying, the entire subtree will be
+    // freed, and the DOM will be handled separately,
+    // so no need to mess with childViews.
+    if (this.isDestroying) { return; }
+
+    // update parent node
+    this.unlinkChild(view);
+
+    // remove view from childViews array.
+    var childViews = get(this, 'childViews');
+
+    var index = childViews.indexOf(view);
+    if (index !== -1) { childViews.splice(index, 1); }
+
+    return this;
+  },
+
+  linkChild(instance) {
+    if (!instance[OWNER]) {
+      setOwner(instance, getOwner(this));
+    }
+
+    instance.parentView = this;
```

Comment:
@stefanpenner `parentView` is slotted. https://github.com/emberjs/ember.js/blob/master/packages/ember-views/lib/views/core_view.js#L69